File: prototype/rails/src/test/domain/reports/status_label_test.rb
require "test_helper"

module Domain
  module Reports
    class StatusLabelTest < ActiveSupport::TestCase
      def test_it_reads_a_single_word_status_as_a_sentence
        assert_equal "Draft", StatusLabel.of("draft")
      end

      def test_it_replaces_the_underscores_of_a_multi_word_status
        assert_equal "For sale", StatusLabel.of("for_sale")
        assert_equal "Awaiting shipment", StatusLabel.of("awaiting_shipment")
        assert_equal "Pending verification", StatusLabel.of("pending_verification")
      end

      def test_it_labels_a_symbol_status
        assert_equal "Paid out", StatusLabel.of(:paid_out)
      end
    end
  end
end

File: prototype/node/src/app/core/status-label.test.ts
import { test } from 'node:test'
import assert from 'node:assert/strict'
import { statusButtons, statusLabel } from './status-label.ts'

test('a single-word status reads as a sentence', () => {
  assert.equal(statusLabel('draft'), 'Draft')
  assert.equal(statusLabel('paid'), 'Paid')
})

test('the underscores of a multi-word status become spaces', () => {
  assert.equal(statusLabel('for_sale'), 'For sale')
  assert.equal(statusLabel('awaiting_shipment'), 'Awaiting shipment')
  assert.equal(statusLabel('pending_verification'), 'Pending verification')
  assert.equal(statusLabel('paid_out'), 'Paid out')
})

test('an empty status reads as nothing', () => {
  assert.equal(statusLabel(''), '')
})

test('each transition becomes a button naming the status it switches to', () => {
  assert.deepEqual(statusButtons(['for_sale', 'archived']), [
    { status: 'for_sale', label: 'Mark for sale' },
    { status: 'archived', label: 'Mark archived' },
  ])
})

test('no legal transitions means no buttons', () => {
  assert.deepEqual(statusButtons([]), [])
})
